infra/modules/env-id/README.md

```diff
@@ -0,0 +1,43 @@
+## Environments
+
+Expectation is that users of terraform modules/examples will want to 'namespace' infra resources
+created by a given Terraform configuration in a manner that groups them together via the interfaces
+of the host platform, making it easy to find all related infrastructure.
+
+To this end, the convention of an `environment_name` variable will be supported.
+
+As this value will be used to prefix infrastructure resource names/ids in many systems with different
+restrictions on allowed characters, we will enforce some universal restrictions on the value, as
+specify some transformations:
+- restrict characters to `[A-Za-z0-9-_ ]`, starting and ending with an alphabetical character
```

Comment:
could default to `w8s-dev` for all new customers when in testing phase, then `w8s-prod`